analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NonDerivedPrivateClassesShouldBeSealed.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class Foo
```

Comment:
Could you please add some tests with `structs` and `records` too? Structs cannot be inherited but records can.